### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by marcogruhl**
handle errors with an exception.
Extended Answer and TwitterException object.

Could you please review if the logic is suitable? I don't like the exception approach, but it is non breaking and simple. As alternative I can imaging some base class for all response objects and parsing like:

```C#
private T[] ParseArrayData<T>(string json) where T : IHaveError
```

I will add some Test this weekend. I couldn't find an exact structure of the `Error` object. I (for now) could only produce errors with an detail**S** property but in the docs i see only a detail property.

I pushed this to discuss these points.

### CODE REVIEW

**Comment by Xwilarg** [comment]
Hey, sorry for the late answer, improving the error handling is definitly a great idea!

But yeah maybe instead it would be better to have many kind of exception instead of having the type as a variable inside?

**Comment by marcogruhl** [comment]
I would like to implement as exact as possible, but i couldn't find docs for the error responses. This is the main doc i used: [Errors](https://developer.twitter.com/en/support/twitter-api/error-troubleshooting). For some endpoints there are examples for the error response like [filtered-stream](https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/integrate/consuming-streaming-data) but only for the fewest. So my implementation is as general as possible:
title, detail, type and errors. In the errors object there are also message, title and type along with details as default and i extend it with parameter/s code and value.
The integration in the Parse method is more like extending the existing `answer.Detail != null` and like i said, non breaking.

I think this is all i can do for now.

**Comment by Xwilarg** [comment]
Uhm yeah I indeed think it's a good idea

**Comment by marcogruhl** [comment]
> Uhm yeah I indeed think it's a good idea

sorry I dont understand, what is a good idea?

I just wanted to say: I cant do better for now.

**Comment by Xwilarg** [comment]
I meant what you are currently doing, I'll check the code more in details in the following days

**Comment by marcogruhl** [comment]
Thank you! No hurry! I can carry on with my depending project!

File: src/Client/TwitterClient.cs [hunk 1]
```diff
@@ -131,9 +131,9 @@ private static void InternalIncludesParse<T>(Answer<T[]> answer)
         private T[] ParseArrayData<T>(string json)
         {
             var answer = JsonSerializer.Deserialize<Answer<T[]>>(json, _jsonOptions);
-            if (answer.Detail != null)
+            if (answer.Detail != null || answer.Errors != null)
             {
-                throw new TwitterException(answer.Detail);
+                throw new TwitterException(answer.Detail ?? "Error. See Errors property.", answer.Title, answer.Type ?? "Error", answer.Errors);
```
Comment (Xwilarg) [inline src/Client/TwitterClient.cs:136]: Maybe all these checks should be moved to the ctor of TwitterException instead?

**Review by Xwilarg — CHANGES_REQUESTED** [review]

**Comment by Xwilarg** [comment]
Hey, should I merge this PR or are you still busy with it?

**Comment by marcogruhl** [comment]
You can merge it. Work is done.

**Pull request merged**